### PR TITLE
Magmoor Digsite IV Rework | Justice for the Lavaplanet™

### PR DESCRIPTION
**Pull request by Overdrive659**
## The Great Rework

Due to "unplayability" and the increasing lack of TGMCs Adderall™ supply, there have been talks of reworking Magmoor Digsite IV. So decided to take matters into my own, crayon stained, hands and here is the result of an arduous MS Paint© project.

Untitled

While many assets of the original map still pertain in the new rendition, Magmoor has been reworked to introduce better gameplay and flow. The map was remade to address certain gameplay issues, one of the main ones being **The Caves.**

win6

In previous versions, Magmoor's northern caves covered ca. 35% of the playable area of the map, and acted more as a playfield division than anything else.

Screenshot_2

The rework addresses this by pushing the caves further north, and instead adding cave passages to the east and west to encouraging a more fluid playstyle by introducing more attack vectors. Leading to xenos being further encouraged on hit and run endeavours, whereas marines can more easily shift a push should their current one bog down.

The second issue the new version tries to solve is the **difficulties surrounding the use of engineering:**

Screenshot_4

Geothermal Generators are a usual contestation point between xenos and marines, leading to heavy fighting in that area. In the original version of Magmoor, the Gens are situated in a single room, below the central fissure. This room, excluding windows, has only 1 exit & entrance which leads to it becoming a binary space, meaning either side controls this room with little conflict. (as apposed to multi-room buildings for example...)

To fix this the rework expands the Geothermal Room upward, adding 3 new walled areas to what used to be a 1 room building:

image

This change encourages a more nuanced fighting over this major point on the map, allowing for forward and backward motion between the 2 sides. On top of this, the rework brings a new light to the central fissure of the map, changing it from a simple obstacle to its own attack lane.

image

The new bridge north of the central fissure adds extra value to the central fissure as a control point and acts as a third main attack vector. _(excluding east & west caves)_ While being centred around the bridge, this chokepoint has variety. The observational decks to the south- east and west allows marines to poke at the xeno defenders, **however should they open up the windows it also opens potential boiler lanes for the hive.**

Other minor changes marked in colour:
- Medbay reworked & moved (Green)
- Engineering workshop added south of Atmos (Yellow)
- Hydroponics merged into service department block, Rest and Recreation (Teal/Cyan)
- New Security Department (Red)
- Reworked Mining (Purple)
- Reworked R&D (Dark Blue)
- New Arrivals "Hallways" (Grey)
- Admin Meeting Room _(Custodial Closet)_ (Black)

## Why is it good for the game?

Magmoor has needed a rework for some time now, the map had gimmicks people seemed to enjoy and I think the map could be made more enjoyable while keeping what made it fun in the first place.

magmoor

## Changelog
:cl:
add: Magmoor Digsite IV Rework
add: Code changes to support extended assets used in rework
del: Removed old Magmoor version.
/:cl:

## Special Thanks To:

grayson

ZARO

### CODE REVIEW

**Comment by RipGrayson** [comment]
This PR has undergone review prior to it becoming a PR, but the config for Magmoor will need to be enabled before it can become a TMC.

**Comment by FosterGrime-Heymynameslouie** [comment]
it clawed its way out of stale pit thanks to you

speede merhg

**Comment by Goblinu** [comment]
[
image
](url)

MAGMOOOOOR!!!!

**Comment by Lumipharon** [comment]
What drives a man to put the 'after' picture before the 'before' picture'? What darkness lurks in their hearts to commit such unspeakable acts?

**Comment by novaepee** [comment]
Not connected. You need a straight line with a distance of less than 15 tiles.
image
image


Not needed. Economy of ai nodes is a good mapping habit.
image

Recommended paths
image

Need AI nodes
image
image
image
image
image
image

Likely places for survivors to go to:
image
image

If you mathematically ensure that survivors cannot survive the walk with fire extinguisher, then it is okay until a survivor finds a better exploit.

I highly recommend for bridges to have fire extinguisher lockers similar to the prior Magmoor map have.
image
image

**Comment by Losenis** [comment]
Alternatively make it so that those openings have a clear shot from boilers.

**Comment by Al-1ce** [comment]
>Likely places for survivors to go to:

those are long removed

**Comment by Overdrive659** [comment]
> image
> 

There is an AI node under the fog tile.
image


> Need AI nodes

Most recent commit adds AI nodes in the places you mentioned and fixes the main bridge AI node issue, plus remaps the ones at research.

> If you mathematically ensure that survivors cannot survive the walk with fire extinguisher, then it is okay until a survivor finds a better exploit.

Removed the eastern cavern, shrank the western lava chasm cavern and counted the tiles. Any survivor would pain-crit before reaching the cave, even if they managed to reach the cave a boiler has a clean shot at them.

**Comment by RipGrayson** [comment]
Actually, I'm going to TMC it even without the config change. Admins can fix the config while it's TMC'ed as well as when it's not.

**Comment by blackdav123** [comment]
zero weeds outside of caves? yep, its magmoor time

**Comment by Overdrive659** [comment]
> Apparently Magmoor is multi-dimensional, this needs to be corrected.

Fixed with latest commit

**Comment by GravitronBlue** [comment]
> Actually, I'm going to TMC it even without the config change. Admins can fix the config while it's TMC'ed as well as when it's not.

- Admin restores Magmoor map config
- PR gets unTM'd
- Server map config not reset
- Old magmoor is now back

This PR was meant to replace the old map not bring it back.

Otherwise hope the changed map plays better once this current mess is fixed up.

File: code/game/turfs/closed.dm [hunk 1]
```diff
@@ -45,6 +45,15 @@
...
 
+/turf/closed/mineral/smoothoutdoor
+	name = "rock"
+	icon = 'icons/turf/walls/lvwall.dmi'
+	icon_state = "lvwall-0-0-0-0"
+	walltype = "lvwall"
+	smoothing_behavior = DIAGONAL_SMOOTHING
+	smoothing_groups = SMOOTH_MINERAL_STRUCTURES
+	open_turf_type = /turf/open/floor/plating/ground/mars/random/dirt
+
```
Comment (RipGrayson) [inline code/game/turfs/closed.dm:56]: It should probably be /turf/closed/mineral/smooth/outdoor
     open_turf_type = /turf/open/floor/plating/ground/mars/random/dirt

there's no need for the other fields, it already inherits those

**Review by RipGrayson — CHANGES_REQUESTED** [review]

**Comment by Overdrive659** [comment]
The PR should be ready for another TM and playtest now:

- Main duplication issue fixed in DME file
- Weed imbalance ironed out
- Small map errors fixed (stairs, light placement, wrongly rotated doors)
- Pipes no longer show through the concrete

The main issue that remains is that the lattices on the map disappear and disposals can't be placed on lava (fluff/decor issue)

File: code/game/turfs/closed.dm (hunk 1 above)
Comment (Overdrive659) [reply code/game/turfs/closed.dm]: check latest

**Comment by Zar000** [comment]
lets goooooo

**Comment by Jalopkomit18** [comment]
invis grille
There are invisible grilles that block movement somewhere in the marked area. I don't know if there are more.

**Comment by Overdrive659** [comment]
>There are invisible grilles that block movement somewhere in the marked area. I don't know if there are more.

For some reason that is BYOND me, the grilles, lattices and a large chunk of the decorative disposals piping disappears when the map loads in game. I guess the colliders stayed behind...? Ill just remove them-

**Comment by Overdrive659** [comment]
Updated the branch and fixed the Merge Conflict, should be ready for a full merge now.

**Comment by Shootnik01** [comment]
What about weed?? Can you add weed spawn with the roundstart?

**Comment by Al-1ce** [comment]
2nding this ^

**Pull request merged**